base/abstractarray.jl

```diff
@@ -385,9 +387,14 @@ eachindex(::LinearFast, A::AbstractArray) = 1:length(A)
 
 function eachindex(A::AbstractArray, B::AbstractArray)
     @_inline_meta
-    eachindex(linearindexing(A)*linearindexing(B), A, B)
+    eachindex(linearindexing(A,B), A, B)
+end
+function eachindex(A::AbstractArray, B::AbstractArray...)
+    @_inline_meta
+    eachindex(linearindexing(A,B...), A, B...)
 end
 eachindex(::LinearFast, A::AbstractArray, B::AbstractArray) = 1:max(length(A),length(B))
```

Comment:
Check out `reducedim.jl` for a fun example.

The idea is that, in cases where sizes differ, you should visit "each index," which means you should essentially traverse the bounding box. `CartesianIndex`es support `min`, so you can safely say `A[min(I, Imin)]` and thereby implement reductions and broadcasting.